src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,582 @@
+                TExpression inlineNode,
+                SyntaxGenerator syntaxGenerator,
+                SyntaxNode root,
+                ImmutableDictionary<ISymbol, SyntaxNode> replacementTable,
+                CancellationToken cancellationToken)
+            {
+                var editor = new SyntaxEditor(inlineNode, syntaxGenerator);
+
+                foreach (var kvp in replacementTable)
+                {
+                    var allReferences = await SymbolFinder
+                        .FindReferencesAsync(kvp.Key, document.Project.Solution, cancellationToken)
+                        .ConfigureAwait(false);
+                    var allSyntaxNodesToReplace = allReferences
+                        .SelectMany(reference => reference.Locations
+                            .Select(location => root.FindNode(location.Location.SourceSpan)))
+                        .ToImmutableArray();
+
+                    foreach (var nodeToReplace in allSyntaxNodesToReplace)
+                    {
+                        if (editor.OriginalRoot.Contains(nodeToReplace))
+                        {
+                            var replacementNodeWithTrivia = kvp.Value
+                                .WithLeadingTrivia(nodeToReplace.GetLeadingTrivia())
+                                .WithTrailingTrivia(nodeToReplace.GetTrailingTrivia());
+                            editor.ReplaceNode(nodeToReplace, replacementNodeWithTrivia);
+                        }
+                    }
+                }
+
+                return editor.GetChangedRoot();
+            }
+
+            /// <summary>
+            /// Generate a dictionary which key is the symbol in Callee (include local variable and parameter), value
+            /// is the replacement syntax node for all its occurence.
+            /// </summary>
+            private static ImmutableDictionary<ISymbol, SyntaxNode> ComputeReplacementTable(
+                AbstractInlineMethodRefactoringProvider<TInvocationNode, TExpression, TArgumentSyntax, TMethodDeclarationSyntax, TIdentifierName>
+                    inlineMethodRefactoringProvider,
+                IMethodSymbol calleeMethodSymbol,
+                ImmutableArray<(IParameterSymbol parameterSymbol, TExpression literalExpression)> parametersWithLiteralArgument,
+                ImmutableArray<IParameterSymbol> parametersWithDefaultValue,
+                SyntaxGenerator syntaxGenerator,
+                ImmutableDictionary<ISymbol, string> renameTable)
+            {
+                // 1. Type arguments (generics)
+                // Example:
+                // Before:
+                // void Caller()
+                // {
+                //     Callee<int>();
+                // }
+                // void Callee<T>() => Print(typeof<T>);
+                // After:
+                // void Caller()
+                // {
+                //     Callee<int>();
+                // }
+                // void Callee<T>() => Print(typeof<int>);
```

Comment:
why is the callee being updted here?  i don't get it.

Reply:
Because if the generics types is specified in the Caller then it also needed to be repalced.
As the example, 'T' is int type so if we don't do the replacement work it will becomes
Before
`void Caller() { Callee<int>(); }`
`Void Callee<T>() { Print(typeof(T)) }`
After:
`Void Caller() { Print(typeof(T))} `
`Void Callee<T>() { Print(typeof(T)) }`

Because T doesn't exist in Caller so the generics is also needed to be replaced.

Reply:
but in your before/after (in the actual comment), the 'caller' code doesn't change.  is that mistake in the comment?